src/mpc_controller.cpp

```diff
@@ -106,7 +108,13 @@ quadrotor_common::ControlCommand MpcController<T>::run(
 
   // Get the feedback from MPC.
   mpc_wrapper_.setTrajectory(reference_states_, reference_inputs_);
-  mpc_wrapper_.update(est_state_, do_preparation_step);
+  if (solve_from_scratch_) {
+    ROS_INFO("Solving MPC from Hover Conditions.");
```

Comment:
This message is missleading.
It only uses hover as initial guess...
additionally, have you tried doing more than one solving step after resetting?
is this tested?